src/clj/ataru/kk_application_payment/kk_application_payment_status_updater_job.clj

```diff
@@ -0,0 +1,234 @@
+          (log/info "Update kk payment status hander for" person-oid application-term application-year)
+          (doseq [payment modified-payments]
+            (let [new-state (:state payment)]
+              (cond
+                (= (:awaiting payment/all-states) new-state)
+                (create-payment-and-send-email job-runner maksut-service payment))))
+
+          (doseq [application-payment existing-payments]
+            (let [{:keys [application payment]} application-payment]
+              (cond
+                (needs-reminder-sent? payment)
+                (send-reminder-email-and-mark-sent job-runner payment application))))
+
+          (invalidate-maksut-payments-if-needed maksut-service modified-payments))
+        (log/debug "Application id" application_id "not in haku with kk application payments")))))
+
+(defn start-update-kk-payment-status-for-person-job
+  [job-runner person-oid term year]
+  (when (get-in config [:kk-application-payments :enabled?])
+    (jdbc/with-db-transaction [conn {:datasource (db/get-datasource :db)}]
+                              (job/start-job job-runner
+                                             conn
+                                             "kk-application-payment-person-status-update-job"
+                                             {:person_oid person-oid :term term :year year}))))
+
+(defn start-update-kk-payment-status-for-application-key-job
+  [job-runner application-key]
+  (when (get-in config [:kk-application-payments :enabled?])
+    (jdbc/with-db-transaction [conn {:datasource (db/get-datasource :db)}]
+                              (job/start-job job-runner
+                                             conn
+                                             "kk-application-payment-person-status-update-job"
+                                             {:application_key application-key}))))
+
+(defn start-update-kk-payment-status-for-application-id-job
+  [job-runner application-id]
+  (when (get-in config [:kk-application-payments :enabled?])
+    (jdbc/with-db-transaction [conn {:datasource (db/get-datasource :db)}]
+                              (job/start-job job-runner
+                                             conn
+                                             "kk-application-payment-person-status-update-job"
+                                             {:application_id application-id}))))
+
+(defn start-update-kk-payment-status-for-all-job
+  [job-runner]
+  (when (get-in config [:kk-application-payments :enabled?])
+    (jdbc/with-db-transaction [conn {:datasource (db/get-datasource :db)}]
+                              (job/start-job job-runner
+                                             conn
+                                             "kk-application-payment-status-update-scheduler-job"
+                                             {}))))
+
+; TODO: we might be updating status of a single person multiple times if they have applications in multiple hakus.
```

Comment:
Jos tätä haluaisi optimoida, voisi palauttaa get-hakus-and-updatelle takaisin aina päivitetyn listan läpikäytyjä (person-oid, term, year) -komboja niin, että seuraavalla kutsulla ne voidaan filtteröidä suoraan pois. Pitäisi riittää, että kukin maksurivi päivitetään vain kerran minkä tahansa haun kontekstissa.